pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -1114,7 +1114,7 @@ func (og *operationGenerator) GenerateMapVolumeFunc(
 
 		}
 		// Call SetUpDevice if blockVolumeMapper implements CustomBlockVolumeMapper
-		if customBlockVolumeMapper, ok := blockVolumeMapper.(volume.CustomBlockVolumeMapper); ok {
+		if customBlockVolumeMapper, ok := blockVolumeMapper.(volume.CustomBlockVolumeMapper); ok && actualStateOfWorld.GetDeviceMountState(volumeToMount.VolumeName) != DeviceGloballyMounted {
```

Comment:
Can you also test the reconstruction scenario (restart kubelet while the volume is already "mounted")?

Reply:
Also the scenario where two pods share the same volume on the same node.

Reply:
There is are tests with `[Testpattern: Pre-provisioned PV (block volmode)] multiVolume [Slow] should concurrently access the single volume from pods on the same node` that passed in pull-kubernetes-e2e-gce-storage-slow.